src/Core/Module/ModuleOverrideVerifier.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Module;
+
+use PrestaShop\PrestaShop\Adapter\Tools;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ModuleOverrideVerifier
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var array
+     */
+    private $errors = [];
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function hasOverrideConflict(string $moduleOverridePath): bool
+    {
+        // if module doesn't have an override folder, there is no conflict
+        if (!is_dir($moduleOverridePath)) {
+            return false;
+        }
+
+        $tools = new Tools();
+        // module has overrides, let's check override files one by one
+        foreach ($tools->scandir($moduleOverridePath, 'php', '', true) as $file) {
+            $moduleOverrideFile = $moduleOverridePath . DIRECTORY_SEPARATOR . $file;
+            $existingOverrideFile = _PS_OVERRIDE_DIR_ . $file;
```

Comment:
The override directory should be injected in the constructor, this will make the code more adaptable and easier to unit test

Reply:
done 👍